internal/tsc_wrapped/compiler_host.ts

```diff
@@ -440,7 +440,7 @@ export class CompilerHost implements ts.CompilerHost, tsickle.TsickleHost {
   writeFile(
       fileName: string, content: string, writeByteOrderMark: boolean,
       onError: ((message: string) => void) | undefined,
-      sourceFiles: ReadonlyArray<ts.SourceFile>): void {
+      sourceFiles: ReadonlyArray<ts.SourceFile> | undefined): void {
```

Comment:
Fixes 
```
external/build_bazel_rules_typescript/internal/tsc_wrapped/compiler_host.ts(446,3): error TS2416: Property 'writeFile' in type 'CompilerHost' is not assignable to the same property in base type 'CompilerHost'.
  Type '(fileName: string, content: string, writeByteOrderMark: boolean, onError: ((message: string) => void) | undefined, sourceFiles: ReadonlyArray<SourceFile>) => void' is not assignable to type 'WriteFileCallback'.
    Types of parameters 'sourceFiles' and 'sourceFiles' are incompatible.
      Type 'ReadonlyArray<SourceFile> | undefined' is not assignable to type 'ReadonlyArray<SourceFile>'.
        Type 'undefined' is not assignable to type 'ReadonlyArray<SourceFile>'.
```
My guess is that in ts 3.1 if a preceding parameter can be undefined (onError in this case) then subsequent parameters are assumed to be allowed to be undefined?

`writeFile` is defined as `type WriteFileCallback = (fileName: string, data: string, writeByteOrderMark: boolean, onError: ((message: string) => void) | undefined, sourceFiles?: ReadonlyArray<SourceFile>) => void;` in typescript 3.1 so its not strictly a call signature mismatch but maybe a change in how call signatures are interpreted?